### PR TITLE
PR: Make file shown in code analysis pane follow the active editor tab

### LINKED ISSUE #1536: Pylint should follow the opened file
_From [contrebasse](https://code.google.com/u/contrebasse/) on 2013-08-26T07:29:21Z_

Spyder Version:  2.3.0dev
Python Version:  2.7.4
Qt Version    :  4.8.4, PyQt4 (API v1) 4.10 on Linux
IPython >=1.0   :  0.13.2 (NOK)
matplotlib >=1.0:  1.2.1 (OK)
pep8 >=0.6      :  1.4 (OK)
pyflakes 0.5.0  :  0.6.1 (NOK)
pylint >=0.25   :  0.26.0 (OK)
rope >=0.9.2    :  0.9.2 (OK)
sphinx >=0.6.6  :  1.1.3 (OK) 

What steps will reproduce the problem?
1. Run a pylint analysis on the current opened file in the editor
2. Change the active file in the editor What is the expected output? What do you see instead? I expect to see the pylint analysis for the new file or a blank analysis if no previous analysis is found. Instead, I still see the analysis for the previous file, which is misleading. 

Please provide any additional information below

. I encountered this problem when trying to improve two modules that work tightly together, and I was constantly switching between the two files.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1536_

### PR DESCRIPTION
**Pull request by neophnx**
## Description of Changes

This implementation introduce a new public slot in order to follow editor tab change. This way filename follows active editor tab without performing code analysis.

Since this is my first contribution to Spyder, I'll appreciate any advice.

* [X] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [X] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

output

### Issue(s) Resolved

Fixes #1536

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
Julien Gosme (neophnx)

### CODE REVIEW

**Comment by goanpeca** [comment]
Hi @neophnx thanks for the contribution. 

Please complete the [text that was there when you opened the PR](https://raw.githubusercontent.com/spyder-ide/spyder/master/.github/PULL_REQUEST_TEMPLATE.md) and add a screenshot/animation of the change using [Licecap](https://www.cockos.com/licecap/) or similar.

**Comment by neophnx** [comment]
Thank you for the direction, I've completed the initial message with the template and added an animated GIF which illustrates to new behaviour.

File: spyder/plugins/pylint/plugin.py [hunk 1]
```diff
@@ -134,6 +138,12 @@ def change_history_depth(self):
         if valid:
             self.set_option('max_entries', depth)
 
+    @Slot()
+    def set_filename(self):
+        """Set filename without code analysis"""
+        self.pylint.set_filename(self.main.editor.get_current_filename())
+
+
```
Comment (CAM-Gerlach) [inline spyder/plugins/pylint/plugin.py]: One blank line between methods.

File: spyder/plugins/pylint/plugin.py [hunk 2]
```diff
@@ -134,6 +138,12 @@ def change_history_depth(self):
         if valid:
             self.set_option('max_entries', depth)
 
+    @Slot()
+    def set_filename(self):
+        """Set filename without code analysis"""
```
Comment (CAM-Gerlach) [inline spyder/plugins/pylint/plugin.py]: Per PEP 257, docstrings must end with a period.

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 3]
```diff
@@ -241,6 +243,15 @@ def analyze(self, filename):
         else:
             self.filecombo.setCurrentIndex(self.filecombo.findText(filename))
         self.filecombo.selected()
+
+    def analyze(self, filename=None):
+        """
+        Code analysis of filename, defaults to the current filename in
+        combobox
```
Comment (CAM-Gerlach) [inline spyder/plugins/pylint/widgets/pylintgui.py]: One line summary should be one line. Also, period.

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 4]
```diff
@@ -231,7 +231,9 @@ def __init__(self, parent, max_entries=100, options_button=None,
         else:
             self.start_button.setEnabled(False)
 
-    def analyze(self, filename):
+    @Slot(str)
+    def set_filename(self, filename):
+        """Set filename without performing code analysis"""
```
Comment (CAM-Gerlach) [inline spyder/plugins/pylint/widgets/pylintgui.py]: Period per PEP 257

**Comment by goanpeca** [comment]
Hi @neophnx jumping in to see if you could update the suggested changes. Thanks!

**Comment by ccordoba12** [comment]
@CAM-Gerlach, this looks good to me. Please make yourself the proposed style changes you mentioned above so we can merge it.

File: spyder/plugins/pylint/plugin.py (hunk 1 above)
Comment (CAM-Gerlach) [reply spyder/plugins/pylint/plugin.py]:
```suggestion
```

File: spyder/plugins/pylint/plugin.py (hunk 2 above)
Comment (CAM-Gerlach) [reply spyder/plugins/pylint/plugin.py]:
```suggestion
        """Set filename without code analysis."""
```

**Comment by CAM-Gerlach** [comment]
@ccordoba12 I would have, but its not just style changes that I can do through Github suggestions. Rather, there are several substantive additions that need to be made, including reverting all the ~hundred whitespace removals (that realistically requires re-doing the changes from scratch without them), making this actually work with split panes (getting the currently active file in the currently active pane), and adding at least a basic test.

**Comment by goanpeca** [comment]
@neophnx could you check this option `allow edits from maintainers`

Screen Shot 2019-05-24 at 23 09 02


So I can help get this PR merged :-)?

**Comment by ccordoba12** [comment]
@goanpeca, if you want to finish this one, you can add @neophnx's clone as a remote, pull it, create a local branch for `pylint-follow-opened-file` and push it again from your clone.

That way @neophnx's commit would be preserved and you could continue working on this.

**Comment by goanpeca** [comment]
>local branch for pylint-follow-opened-file and push it again from your clone.

Would that push to this same PR, or you meant I create a new PR preserving his changes ?

**Comment by ccordoba12** [comment]
> you meant I create a new PR preserving his changes ?

Since @neophnx didn't gave us permissions to write here, you need to create a new PR. And my procedure above will preserve @neophnx's commits.

**Comment by goanpeca** [comment]
Ok! will do

**Comment by ccordoba12** [comment]
Ok, thanks!! Please close this PR when the new one is in place.

**Comment by goanpeca** [comment]
Superseeded by https://github.com/spyder-ide/spyder/pull/9430/files

**Comment by CAM-Gerlach** [comment]
> you can add @neophnx's clone as a remote, pull it, create a local branch for pylint-follow-opened-file and push it again from your clone.

Or even easier, assuming you have the standard ``git pr`` alias setup, you can just do ``git pr 9238`` (if you haven't already to test the branch) and rename it to whatever you want.

**Pull request closed without merging**